rca/shortcourses/models.py

```diff
@@ -54,4 +72,58 @@ class ShortCoursePage(BasePage):
             heading=_("Course Introduction"),
         ),
         StreamFieldPanel("about"),
+        FieldPanel("access_planit_course_id"),
+        MultiFieldPanel(
+            [
+                FieldPanel("course_details_text"),
+                SnippetChooserPanel("frequently_asked_questions"),
+                SnippetChooserPanel("terms_and_conditions"),
+            ],
+            heading="course details",
+        ),
     ]
+
+    def get_access_planit_data(self):
+        data = AccessPlanitXML(course_id=self.access_planit_course_id)
+        return data.get_data()
+
+    def _format_booking_bar(self, access_planit_data, register_interest_link):
+        """ Booking bar messaging with the next course data available
+        Find the next course date marked as status='available' and advertise
+        this date in the booking bar. If there are no courses available, add
+        a default message."""
+
+        booking_bar = {
+            "message": "Applications are now closed",
+            "action": "Register your interest for upcoming dates",
+        }
+        # If there are no dates the booking link should go to a form, not open
+        # a modal, this link is also used as a generic interest link too though.
+        if self.access_planit_course_id:
+            booking_bar["link"] = register_interest_link
+
+        if access_planit_data:
+            for key, date in enumerate(access_planit_data):
+                if date["status"] == "Available":
+                    booking_bar["message"] = "Next course starts"
```

Comment:
Yea this is intended. I only want to change this data `if date["status"] == "Available"`, otherwise the earlier booking_bar definition should be used.